ci-operator/step-registry/ingress/aws/conf/subnets/ingress-aws-conf-subnets-commands.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+# For disconnected or otherwise unreachable environments, we want to
+# have steps use an HTTP(S) proxy to reach the API server. This proxy
+# configuration file should export HTTP_PROXY, HTTPS_PROXY, and NO_PROXY
+# environment variables, as well as their lowercase equivalents (note
+# that libcurl doesn't recognize the uppercase variables).
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# output example:
+# public_subnets: "ci-op-0r5slpp6-0dd5e-8r4pp-subnet-public-us-east-1a","ci-op-0r5slpp6-0dd5e-8r4pp-subnet-public-us-east-1d"
+
+public_subnets=$(oc -n openshift-machine-api get machinesets.machine.openshift.io -ojsonpath='{.items[*].spec.template.spec.providerSpec.value.subnet.filters[].values}' | tr -d "[]" | sed 's/ /,/g' | sed 's/subnet-private/subnet-public/g')
+echo "public subnets: $public_subnets"
+
+lb_type=$(oc get ingress.config cluster -ojsonpath='{.spec.loadBalancer.platform.aws.type}')
+echo "load balancer type: $lb_type"
+
+if [[ "$lb_type" = "Classic" ]]; then
+    oc -n openshift-ingress-operator patch ingresscontroller/default --type=merge -p '{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"providerParameters":{"type":"AWS","aws":{"type":"Classic","classicLoadBalancer":{"subnets":{"ids":null,"names":['"$public_subnets"']}}}},"scope":"External"}}}}'
+fi
+
+if [[ "$lb_type" = "NLB" ]]; then
+    oc -n openshift-ingress-operator patch ingresscontroller/default --type=merge -p '{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"providerParameters":{"type":"AWS","aws":{"type":"NLB","networkLoadBalancer":{"subnets":{"ids":null,"names":['"$public_subnets"']}}}},"scope":"External"}}}}'
+fi
+
```

Comment:
To be robust, suggest to add a exit step if the type of lb is neither `Classic`, nor `NLB`.

Reply:
Using `case` statement to check and `exit 1` if not matched.